runtime/src/main/java/org/corfudb/runtime/view/LayoutManagementView.java

```diff
@@ -194,6 +196,42 @@ public boolean mergeSegments(Layout currentLayout)
         return true;
     }
 
+    /**
+     * Best effort attempt to removes a node from the layout.
+     *
+     * @param currentLayout the layout to remove the node from
+     * @param endpoint the node to remove
+     */
+    public void removeNode(@Nonnull Layout currentLayout,
+                           @Nonnull String endpoint) throws QuorumUnreachableException,
+            OutrankedException {
+
+        Layout newLayout;
+        if (currentLayout.getAllServers().contains(endpoint)) {
```

Comment:
getAllServers does not consider the unresponsive servers list.
Assume an endpoint is only a log unit node. On failure, it will be removed from the chain and added into the unresponisveServers list.

Reply:
Then it should be called getAllActiveServers, will rename it and create a getAllServers method too.

Reply:
Done.